lib/ecto/repo/schema.ex

```diff
@@ -677,14 +677,22 @@ defmodule Ecto.Repo.Schema do
     # already filtered the results to be only about fields.
     data =
       changeset.data
-      |> Map.merge(changes)
+      |> merge_changes(changes)
       |> Map.merge(embeds)
       |> merge_autogen(autogen)
       |> apply_metadata(state, schema_meta)
       |> load_each(values, types, adapter)
     Map.put(changeset, :data, data)
   end
 
+  defp merge_changes(data, changes) do
+    Enum.reduce(changes, data, fn {key, value}, data ->
```

Comment:
The most efficient approach is actually to remove any unknown change and then still call `Map.merge`  at the end, as it requires less modifications. :)

Reply:
so something like:
```
changes = Enum.reduce(changes, changes, fn {key, value}, acc -> if not in key remove from acc end)
Map.merge(data, changes)
```